### PR TITLE
Add axis and spacing to list.

### PR DESCRIPTION
**Pull request by derekdreery**
Add spacing and axis options to list widget. Spacing can be a value or resolved from env.

Closes #1364.

### CODE REVIEW

File: druid/src/widget/list.rs [hunk 1]
```diff
@@ -23,26 +24,54 @@ use crate::im::Vector;
...
     closure: Box<dyn Fn() -> Box<dyn Widget<T>>>,
     children: Vec<WidgetPod<T, Box<dyn Widget<T>>>>,
+    axis: Axis,
+    spacing: KeyOrValue<f64>,
 }
 
 impl<T: Data> List<T> {
     /// Create a new list widget. Closure will be called every time when a new child
     /// needs to be constructed.
-    pub fn new<W: Widget<T> + 'static>(closure: impl Fn() -> W + 'static) -> Self {
```
Comment (cmyr) [inline druid/src/widget/list.rs:39]: Maybe this is heresy, but I think I would leave `new` as is (it creates a vertical list) and then just have a `horizontal` constructor that makes a horizontal list; I think the vertical case is going to be 95%+ of uses cases.

**Review by cmyr — COMMENTED** [review]
Looks good! One thought on the API and one note below, but otherwise I'm happy for this to get in; let me know what you think.

For max correctness one other thing we might want to do now is to check in `layout` if the spacing key has changed in the env, and then call `request_update` if it has.

File: druid/src/widget/list.rs (hunk 1 above)
Comment (derekdreery) [reply druid/src/widget/list.rs:39]: Yep can do that.

**Comment by derekdreery** [comment]
> For max correctness one other thing we might want to do now is to check in layout if the spacing key has changed in the env, and then call request_update if it has.

I suspect that this should be added to other widgets as well. Shall I have a look through?

**Comment by cmyr** [comment]
yes, the ability to inspect changed keys is relatively new (7db6dd01) and this definitely isn't being done optimally by a bunch of widgets.

**Review by cmyr — APPROVED** [review]
I was thinking that `horizontal` would be a constructor? We could document this in `new`, and say something to the effect, "To construct a horizontal list, use [`List::horizontal`] instead?

I'm open to discussion here, but this is at least what I'd had in mind.

In any case, I'm going to approve; feel free to merge either as-is, or with that change.

**Comment by derekdreery** [comment]
I guess it just felt more natural to me, if there weren't two constructors with equal weighting, then the axis should be a parameter rather than in the constructor.

**Comment by derekdreery** [comment]
I will try to have a look at the `KeyOrValue` thing more comprehensively in a separate pass.

**Pull request merged**